src/sortStudents.ts

```diff
@@ -1,16 +1,89 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: [], sortBy: SortType, order: SortOrder,
+): Student[] {
+  function getAverageGrade(pupils: Student): number {
+    return pupils.grades.reduce((sum: number, curr: number) => sum + curr)
+    / pupils.grades.length;
+  }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  const studentsCopy: Student[] & {averageGrade?: number} = [...students];
```

Comment:
```suggestion
  const studentsCopy: Student[] = [...students];
```
Why add {averageGrade?: number} to the copy of the array?